hw/xbox/xid-sdl.c

```diff
@@ -512,14 +512,16 @@ static void usb_xbox_gamepad_realize(USBDevice *dev, Error **errp)
 #endif
 
 #ifdef FORCE_FEEDBACK
-    s->sdl_haptic = SDL_HapticOpenFromJoystick(s->sdl_gamepad);
+    s->sdl_haptic = SDL_HapticOpen(s->device_index);
```

Comment:
`SDL_HapticOpenFromJoystick` segfaults. Not sure why, SDL2 docs suggest you can use a gamepad like a joystick, gamepad just has convenient button mappings.

This works for now, but reminder for myself to try and get `s->sdl_gamepad` to work as I feel that is a slightly cleaner impl, maybe needs a cast?

Reply:
A GameController internally uses a joystick, which you have to retrieve with [SDL_GameControllerGetJoystick](https://wiki.libsdl.org/SDL_GameControllerGetJoystick).

---

Ideally we'd do it the other way around: use a more powerful Joystick, and then only use GameController API to find a good default mapping - this was my original design but it was unfortunately changed when my code was ported to QEMU 2.x.
This redesign is obviously out of scope for this PR.